keras_cv/models/object_detection/yolox/__internal__/binary_crossentropy.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+class BinaryCrossentropy(tf.keras.losses.Loss):
```

Comment:
Instead of having to create our own copy of BinaryCrossentropy, as an interim could we just:

- Add a call to `tf.expand_dims` to the output of YOLOX (probably in the yolox_head)
- Use `axis=0` in the call to the existing BinaryCrossentropy?

(Sorry if I am misunderstanding -- just trying to see if there's a workaround that doesn't require this fork of BinaryCrossentropy until https://github.com/keras-team/keras/issues/17207 is resolved)

Reply:
I briefly discussed this alternative with @LukeWood and it should work.
Should I add this workaround instead of a new internal loss?

Reply:
IMO the workaround is probably nicer, since it's a lot less code and we can put a comment on it explaining what it's doing and why.

That said, I am not strongly opinionated and I am fine with whichever you and @LukeWood prefer

Reply:
What is the purpose of this extra implementation? Is this for label smoothing?
In general, we need to reason more about re-implementing a loss which is already supported by Keras